packages/venia-concept/src/components/CategoryList/categoryList.js

```diff
@@ -66,30 +77,51 @@ class CategoryList extends Component {
 
                         return (
                             <div className={classes.content}>
-                                {data.category.children.map((item, index) => (
-                                    <Link
-                                        className={classes.item}
-                                        to={`/${
-                                            item.url_key
-                                        }${categoryUrlSuffix}`}
-                                        key={index}
-                                    >
-                                        <span className={classes.imageWrapper}>
-                                            {item.image && (
-                                                <img
-                                                    className={classes.image}
-                                                    src={`/pub/media/catalog/category/${
-                                                        item.image
-                                                    }`}
-                                                    alt={item.name}
-                                                />
-                                            )}
-                                        </span>
-                                        <span className={classes.name}>
-                                            {item.name}
-                                        </span>
-                                    </Link>
-                                ))}
+                                {data.category.children.map((item, index) => {
+                                    let imagePath;
+                                    if (item.image) {
+                                        imagePath = makeCategoryMediaPath(
+                                            item.image
+                                        );
+                                    } else {
+                                        const firstProduct =
+                                            item.productImagePreview.items[0];
+                                        if (
+                                            firstProduct &&
+                                            firstProduct.small_image
+                                        ) {
+                                            imagePath = makeProductMediaPath(
+                                                firstProduct.small_image.path
+                                            );
+                                        }
+                                    }
+                                    return (
+                                        <Link
+                                            className={classes.item}
+                                            to={`/${
```

Comment:
The same, we can move this logic to helpers, we can add helper function `buildItemUrl`